### PR TITLE
Functional tests - Update linkchecker for playwright

### PR DESCRIPTION
**Pull request by boubkerbribri**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | 1.7.7.x
| Description?  | Update linkchecker to cover more URL (in 2 languages) and more errors (tinyMCE ones for exemple)
| Type?         | refacto
| Category?     | TE
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | no
| How to test?  | `URL_FO=shopUrl/ npm run linkchecker` 

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/19727)

### CODE REVIEW

File: tests/UI/campaigns/tools/linkchecker.js [hunk 1]
```diff
@@ -133,8 +170,8 @@ async function crawlPage(browserPage, thisPageToCrawl) {
  */
 async function checkResponseStatus(url, status) {
   if (status.startsWith('4') || status.startsWith('5')) {
-    requestError = true;
-    requestTextError = `Request error : ${url} (${status}`;
+    responseError = true;
+    responseTextError = `Request error : ${url} (${status}`;
```
Comment (NeOMakinG) [inline tests/UI/campaigns/tools/linkchecker.js]: Is it normal that parenthesis are not closed?

File: tests/UI/campaigns/tools/linkchecker.js [hunk 2]
```diff
@@ -24,15 +24,18 @@ let outputEntry = {
   url: '',
   passed: [],
   failed: [],
-  jsError: [],
+  Error: [],
```
Comment (PierreRambaud) [inline tests/UI/campaigns/tools/linkchecker.js]:
```suggestion
  errored: [],
```

File: tests/UI/campaigns/tools/linkchecker.js [hunk 3]
```diff
@@ -57,14 +60,30 @@ describe('Crawl every page for defects and issues', async () => {
...
+
+
+    // Intercept console errors
+    await page.on('console', (msg) => {
+      // Handle only errors.
+      if (msg.type() === 'error') {
+        consoleTextError = msg.text();
+        consoleError = true;
+
+        outputEntry.Error.push({
```
Comment (PierreRambaud) [inline tests/UI/campaigns/tools/linkchecker.js]:
```suggestion
        outputEntry.errored.push({
```

File: tests/UI/campaigns/tools/linkchecker.js [hunk 4]
```diff
@@ -57,14 +60,30 @@ describe('Crawl every page for defects and issues', async () => {
...
     // Intercepts JS errors
-    await page.on('pageerror', (pageerr) => {
-      javascriptTextError = pageerr.toString();
+    await page.on('pageerror', (exception) => {
+      javascriptTextError = exception.toString();
       javascriptError = true;
 
-      outputEntry.jsError.push({
-        error: javascriptTextError,
+      outputEntry.Error.push({
```
Comment (PierreRambaud) [inline tests/UI/campaigns/tools/linkchecker.js]:
```suggestion
      outputEntry.errored.push({
```

File: tests/UI/campaigns/tools/linkchecker.js [hunk 5]
```diff
@@ -102,15 +138,16 @@ describe('Crawl every page for defects and issues', async () => {
...
   javascriptError = false;
+  consoleError = false;
 
   outputEntry = {
     name: thisPageToCrawl.name,
     url: thisPageToCrawl.url,
     passed: [],
     failed: [],
-    jsError: [],
+    Error: [],
```
Comment (PierreRambaud) [inline tests/UI/campaigns/tools/linkchecker.js]:
```suggestion
    errored: [],
```

File: tests/UI/campaigns/tools/linkchecker.js (hunk 1 above)
Comment (boubkerbribri) [reply tests/UI/campaigns/tools/linkchecker.js]: shouldn't be a parenthesis, nice catch

**Review by NeOMakinG — APPROVED** [review]

File: tests/UI/campaigns/tools/linkchecker.js (hunk 1 above)
Comment (boubkerbribri) [reply tests/UI/campaigns/tools/linkchecker.js]: sorry it was parenthesis :laughing:

**Review by PierreRambaud — APPROVED** [review]

**Pull request merged**